### PR TITLE
[Issue 1950] Do not remove whitespace after number

### LINKED ISSUE #1950: Do not remove whitespace after number
# Description
Whitespace after number have been removed, which will case an `Uncaught SyntaxError: Invalid or unexpected token` error.

# Input
The code looked like this before beautification:
```
1000000000000000100 .toFixed(0)!=="1000000000000000128")
```

# Expected Output
The code should have looked like this after beautification:
```
1000000000000000100 .toFixed(0) !== "1000000000000000128")
```

# Actual Output
The  code actually looked like this after beautification:
```
1000000000000000100.toFixed(0) !== "1000000000000000128")
```

# Steps to Reproduce

## Environment
OS:

### PR DESCRIPTION
**Pull request by bhaveshvasnani**
# Description
- [x] Source branch in your fork has meaningful name (not `main`)

Fixes Issue: 
Fixes #1950 causing `Uncaught SyntaxError: Invalid or unexpected` token by checking if the input before the dot is a number or not. Depending on the regex used to perform a number match, we set space_before_token to true in the handle_dot function.

As an additional work since the expectation was for both js as well python beautifier, I went ahead and tried to made appropriate python changes. But the python implementation for the same fails to compile as the latest version of black has removed support for py27. https://pypi.org/project/black/. Therefore, I have also included the whitespacing changes that were made by the latest version of black and has also pyproject.toml to remove the dependency.

Error reported before removing dependency for py27 for black:
```
Testing python beautify functionality...
/mnt/c/Open_Source/js-beautify/tools/python-dev3 black --config=python/pyproject.toml python
Usage: black [OPTIONS] SRC ...
Try 'black -h' for help.

Error: Invalid value for '-t' / '--target-version': 'py27' is not one of 'py33', 'py34', 'py35', 'py36', 'py37', 'py38', 'py39', 'py310'.
make: *** [Makefile:33: py] Error 2
```

Have verified that all the tests are passing after this change.

# Before Merge Checklist 
These items can be completed after PR is created.

(Check any items that are not applicable (NA) for this PR)

- [x] JavaScript implementation
- [x] Python implementation (NA if HTML beautifier)
- [x] Added Tests to data file(s)
- Added command-line option(s) (NA if
- README.md documents new feature/option(s)

### CODE REVIEW

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Still needs tests.  I think this code will fail on the following input `a.b21.send()`.

**Comment by bhaveshvasnani** [comment]
> Still needs tests. I think this code will fail on the following input `a.b21.send()`.

Thanks!! Will add tests for this change. Also, the input of type `a.b21.send()` works and the result is the same. Is that not the expectation for such cases?

File: test/data/javascript/tests.js [hunk 1]
```diff
@@ -4546,6 +4546,15 @@ exports.test_data = {
             '    }',
             ')'
           ]
+        },
+        {
+          comment: 'Issue #1950: Do not remove whitespace after number - test scenario: number before a dot',
```
Comment (bhaveshvasnani) [inline test/data/javascript/tests.js:4551]: I'm getting a test failed error here even when I see that expected and ouput are the same. @bitwiseman, any idea what could I be missing?
`Testing js-beautifier with node.js CommonJS (src/index)...
==== Destructured and related ============================================================
---- input -------
1000000000000001000 .toFixed(0)!==1000000000000001024
---- expected ----
1000000000000000100 .toFixed(0) !== 1000000000000001024
---- output ------
1000000000000001000 .toFixed(0) !== 1000000000000001024
---- expected-ws ------
1000000000000000100_.toFixed(0)_!==_1000000000000001024
---- output-ws ------
1000000000000001000_.toFixed(0)_!==_1000000000000001024
================================================================`

**Comment by bitwiseman** [comment]
Sorry, What I meant was `a.b21 . performAction()`. The space should be removed but it looks like this change woy preserve it.

**Comment by bhaveshvasnani** [comment]
> Sorry, What I meant was `a.b21 . performAction()`. The space should be removed but it looks like this change would preserve it.

@bitwiseman, so I have used refex to match numbers from the start to the end. Therefore, a.b21 . performAction() would be modified to remove space accordingly to produce. a.b21.performAction(). I have also modified the test to represent this scenario.

**Comment by bhaveshvasnani** [comment]
@bitwiseman, I'm unable to root cause why 'make ci' is failing for me. Any pointers to understand what could be wrong?

``` ----------------------------------------
Testing js-beautify cli behavior...
Created /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw...
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-file.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-pipe.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-n.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-rn.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-eof.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-eof2.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-n-dash.js
beautified /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-rn-dash.js
Files /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-n-dash.js and /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-rn-dash.js differ
Files /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-n.js and /mnt/c/OpenSource/js-beautify/python/target/test_temp_v8Gw/js-beautify-rn.js differ
unbalanced parenthesis at position 0
js-beautify output for 'src/*.js' was expected have 2 files.
unbalanced parenthesis at position 0```

File: js/src/javascript/beautifier.js [hunk 2]
```diff
@@ -1425,6 +1425,10 @@ Beautifier.prototype.handle_dot = function(current_token) {
     this.handle_whitespace_and_comments(current_token, true);
   }
 
+  if (this._flags.last_token.text.match('^[0-9]+$')) {
```
Comment (bitwiseman) [inline js/src/javascript/beautifier.js:1428]: Ah, I see.

**Review by bitwiseman — APPROVED** [review]
Ah, I get it.  Thanks for modifying the test.

**Comment by bitwiseman** [comment]
@bhaveshvasnani
Not sure why `make ci` is failing for you.  Usually when I can't nail down a problem, the first thing I try is getting a clean, up-to-date environment: make sure everything I care about is checked in, `git clean -xfd`, `git reset --hard`, `git pull`, `git merge upstream/main`, etc. Then try the build again.

**Comment by bhaveshvasnani** [comment]
> @bhaveshvasnani Not sure why `make ci` is failing for you. Usually when I can't nail down a problem, the first thing I try is getting a clean, up-to-date environment: make sure everything I care about is checked in, `git clean -xfd`, `git reset --hard`, `git pull`, `git merge upstream/main`, etc. Then try the build again.

Thank you @bitwiseman!! Let me see if I could get make ci build clean.

**Comment by bhaveshvasnani** [comment]
> @bhaveshvasnani Not sure why `make ci` is failing for you. Usually when I can't nail down a problem, the first thing I try is getting a clean, up-to-date environment: make sure everything I care about is checked in, `git clean -xfd`, `git reset --hard`, `git pull`, `git merge upstream/main`, etc. Then try the build again.

@bitwiseman so the actual problem was I was incorrectly using regex search function and sorting that out fixed make ci locally for. So, hopefully it should be good now.

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**